src/pyFAI/integrator/common.py

```diff
@@ -1658,21 +1658,29 @@ def __save2D(self, filename, intensity, dim1, dim2, error=None, dim1_unit=units.
 # Some properties
 ################################################################################
 
-    def set_darkcurrent(self, dark):
-        self.detector.set_darkcurrent(dark)
+    @property
+    def darkcurrent(self):
+        return self.detector.darkcurrent
 
-    def get_darkcurrent(self):
-        return self.detector.get_darkcurrent()
+    @darkcurrent.setter
+    def darkcurrent(self, dark):
+        self.detector.darkcurrent = dark
+
+    get_darkcurrent = deprecated(darkcurrent.fget, reason="use property", since_version="2025.09")
+    set_darkcurrent = deprecated(darkcurrent.fset, reason="use property", since_version="2025.09")
 
-    darkcurrent = property(get_darkcurrent, set_darkcurrent)
 
-    def set_flatfield(self, flat):
-        self.detector.set_flatfield(flat)
 
-    def get_flatfield(self):
-        return self.detector.get_flatfield()
+    @property
+    def flatfield(self):
+        return self.detector.flatfield
+
+    @flatfield.setter
+    def set_flatfield(self, flat):
```

Comment:
The pre-commit hook spotted a bug at this place and the name of the method should be `flatfield`, not `set_flatfield`